<center> <h1> <b>HOMEWORK 1<b></h1> </center>
<center> <h2> Numerical Analysis II - 2022</h2> </center>
<center> <h3> Fabián Eduardo Suárez Castellanos and Sebastian Aguilera Novoa</h3> </center>

In [17]:
using QuadGK, Calculus, Roots, LaTeXStrings, PlotlyJS, Images

## 1. T

Show that the space $X = \left\{ u\in C^1([-1,1]):u(-1)=u(1)=0 \right\}$, equipped with the norm

$$||u||_X^2 = \int_0^1 (u(x)^2+u'(x)^2)\; dx$$
is not a Banach space. (Hint: Prove that given the sequence $u_n(x) = \sqrt{1+\frac{1}{n}} - \sqrt{x^2+\frac{1}{n}} $, $x\in[-1,1]$, converges to a function $u\notin X$

$$ X=\{u \in C^1[-1,1]: u(-1)=u(1)=0\}$$

$$||u||_x^2=\int_0^1(u(x))^2+(u'(x))^2dx$$


We are going to show that the limit of the sequence $u_n(x)=\sqrt{1 + \frac{1}{n}}-\sqrt{x^2 + \frac{1}{n}}$ is not in $X$ $\Big($that is, it does not converge in $X\Big)$, even though the sequence is Cauchy in $X$.

### Solution

Let $\epsilon > 0$ be, we are going to find $N \in\mathbb N$ such that for every $n,m > N \quad \Rightarrow \quad $   $||u_n(x)-u_m(x)||_X < \epsilon$

Suppose $m > n$ $\Rightarrow$ $\frac{1}{m}<\frac{1}{n}$ $\Rightarrow$ $x^2 + \frac{1}{m} < x^2+ \frac{1}{n}$ $\Rightarrow$ $\sqrt{x^2 + \frac{1}{m}} < \sqrt{ x^2+ \frac{1}{n}}$ and similarly $\sqrt{1 + \frac{1}{m}} < \sqrt{ 1 + \frac{1}{n}}$ then :

$$ ||u_n(x)-u_m(x)||_x < \left|\left|\left(\sqrt{1 + \frac{1}{n}}-\sqrt{x^2 + \frac{1}{n}} \right)-\left(\sqrt{1 + \frac{1}{m}}-\sqrt{x^2 + \frac{1}{m}}\right)\right|\right|_X \\
<\left|\left|\sqrt{1 + \frac{1}{n}}-\sqrt{x^2 + \frac{1}{n}}\right|\right|_X < \left|\left|\sqrt{1 + \frac{1}{n}}\right|\right|_X $$

since  $x \in [-1,1]$

$$ \left|\left|\sqrt{1 + \frac{1}{n}}\right|\right|_X = \left(\int_0^1\left(\sqrt{1 + \frac{1}{n}}\right)^2+(0)^2)dx\right)^{1/2} = \left(\int_0^1\left(\sqrt{1 + \frac{1}{n}} \right)^2dx\right)^{1/2} \\
= \left(\int_0^1\left(1 + \frac{1}{n}\right) dx\right)^{1/2} = \left(1 + \frac{1}{n}\right)^{1/2} $$

Then,

$$ \left(1 + \frac{1}{n}\right)^{1/2} < \left(1 + \frac{1}{N}\right)^{1/2} < \epsilon \\
1 + \frac{1}{N} < \epsilon ^2 \quad \leftrightarrow \quad \frac{1}{N} < \epsilon ^2 - 1 \qquad \Longrightarrow \quad N > \frac {1}{\epsilon ^2 - 1} $$

Taking  $N > \frac {1}{\epsilon ^2 - 1}$ we have proved that $u_n(x)$  is the cauchy and convergent and $\lim_{n \to \infty}(u_n(x))= 1-|x|= u(x) $ but $u(x)$ does not converge in  $X$ because its firts derivate is not continuous in $ [-1,1]$ . 

Hence, $X$ is not complete and therefore $X$ is not a Banach space. 

In [69]:
x = range(-1, 1, step=0.1) |> collect
u(n,x) = sqrt(1+1/n) -sqrt(x^2+1/n)

pp0 = scatter(;x=x, y=u.(10^0,  x),  mode="lines+markers", name="u(10^0.5, x)")
pp1 = scatter(;x=x, y=u.(10^0.5,x),  mode="lines+markers", name="u(10^0.5, x)")
pp2 = scatter(;x=x, y=u.(10^1,  x),  mode="lines+markers", name="u(10^1,   x)")
pp3 = scatter(;x=x, y=u.(10^1.5,x),  mode="lines+markers", name="u(10^1.5, x)")
pp4 = scatter(;x=x, y=u.(10^2,  x),  mode="lines+markers", name="u(10^2,   x)")
pp5 = scatter(;x=x, y=u.(10^2.5,x),  mode="lines+markers", name="u(10^2.5, x)")
pp6 = scatter(;x=x, y=u.(10^3,  x),  mode="lines+markers", name="u(10^3,   x)")
pp7 = scatter(;x=x, y=u.(10^4,  x),  mode="lines+markers", name="u(10^4,   x)")
pp8 = scatter(;x=x, y=u.(10^5,  x),  mode="lines+markers", name="u(10^5,   x)")
ppx = scatter(;x=x, y=-abs.(x).+1,  mode="lines+markers", name="1-|x|")

plots = [pp0, pp1, pp2, pp3, pp4, pp5, pp6, pp7, pp8, ppx]
pp1 = plot(plots, Layout(title="Un Sequence",  xaxis_title="x",  yaxis_title="u_n"))


data: [
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y",
  "scatter with fields mode, name, type, x, and y"
]

layout: "layout with fields margin, title, xaxis, and yaxis"

## 2. T

Let $u(x)$ and $v(x)$ two functions defined as follows

$$ 
u(x)=\left\{\begin{array}{ll}
x, & \text { if } 0 \leq x \leq 1 \\
2-x & \text { if } 1 \leq x \leq 2
\end{array} \quad y \quad v(x)=\sin \pi x\right.
$$

with their derivatives

$$ 
u'(x)=\left\{\begin{array}{ll}
1, & \text { if } 0 \leq x \leq 1 \\
-1 & \text { if } 1 \leq x \leq 2
\end{array} \quad y \quad v'(x) = \pi\cos \pi x\right.
$$

In [2]:
function u(x)
    if 0 <= x <=1
        return x
    elseif 1 < x <=2
        return 2 - x
    else 
        return 0
    end
end

function up(x)
    if 0 <= x <=1
        return 1
    elseif 1 < x <=2
        return -1
    else 
        return 0
    end
end

v(x)  = sin(pi * x)
vp(x) = pi * cos(pi * x)

x = range(0, 2, step=0.1) |> collect;

In [55]:
plot_u  = scatter(;x=x, y=u.(x), mode="lines+markers", name="u(x)")
plot_v  = scatter(;x=x, y=v.(x), mode="lines+markers", name="v(x)")
plot_uv = scatter(;x=x, y=v.(x).*u.(x), mode="lines+markers", fill="tozeroy", name="u(x) * v(x)")

plot_up = scatter(;x=x, y=up.(x), mode="lines+markers", name="u'(x)")
plot_vp = scatter(;x=x, y=vp.(x), mode="lines+markers", name="v'(x)")
plot_uvp = scatter(;x=x, y=vp.(x).*up.(x), mode="lines+markers", fill="tozeroy", name="u'(x) * v'(x)")

p1 = plot([plot_u, plot_v,plot_uv], Layout(title="Functions",  xaxis_title="x"))
p2 = plot([plot_up, plot_vp,plot_uvp], Layout(title="Function Derivative",  xaxis_title="x"))

p12 = [p1 p2]
#relayout!(p, height=300, width=700, title_text="Functions", legend_title_text="Legend")
#p

data: [
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields fill, mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields fill, mode, name, type, x, xaxis, y, and yaxis"
]

layout: "layout with fields annotations, margin, xaxis1, xaxis2, yaxis1, and yaxis2"

### Orthogonality 


#### - $L^2(0,2)$ Space

Norm of $L^2$ and its norm $||u||_{L^2}$

$$ (u,v) =  \int_0^2 u(x) v(x) dx  = \int_0^1 x \; \sin(\pi x) dx + \int_1^2 (2-x)\; \sin(\pi x) dx $$

Integranting by steps with the sustitution $w = x$ and $dz = \sin (\pi x) \; dx$

$$ (u,v)_{L^2} = \frac{1}{\pi}x\cos(\pi x)\Big|_0^1 - \frac{1}{\pi}\int_0^1 \; \cos(\pi x) dx + \frac{1}{\pi}(2-x)\cos(\pi x)\Big|_1^2 - \frac{1}{\pi} \int_1^2 \; \cos(\pi x) dx \\
(u,v)_{L^2} = \frac{1}{\pi}\cos(\pi) - \frac{1}{\pi^2}\sin(\pi x)\Big|_0^1 - \frac{1}{\pi}\cos(\pi) - \frac{1}{\pi^2}  \sin(\pi x)\Big|_0^1  \\
(u,v)_{L^2} = \frac{1}{\pi}\cos(\pi) - \frac{1}{\pi}\cos(\pi) -  \frac{1}{\pi^2} ( \sin(\pi) - \sin(0) ) - \frac{1}{\pi^2} ( \sin(2\pi) - \sin(\pi) ) = \frac{1}{\pi} - \frac{1}{\pi} -(0-0) -(0-0) \\
\Longrightarrow  \quad \boxed{(u,v)_{L^2} =  0}
$$
The function $u$ and $v$ are orthogonals in $L^2(0,2)$.  

In [4]:
integral, err = quadgk(x -> u(x)*v(x), 0, 2, rtol=1e-8)

(2.0816681711721685e-17, 0.0)

#### - $H^1(0,2)$ Space

Now, let's check if $u$ and $v$ are orthogonal in $H^1(0,2)$ usign the its norm and their derivatives

$$ 
(u,v)_{H^1} =  \int_0^2 u(x) v(x) dx  + \int_0^2 u'(x) v'(x) dx = 0 + \int_0^1 u'(x) v'(x) dx  + \int_1^2 u'(x) v'(x) dx  \\
(u,v)_{H^1} = \pi\int_0^1 1 \; \cos(\pi x) dx + \pi\int_1^2 (-1)\; \cos(\pi x) dx = \sin(\pi x) \Big|_0^1 - \sin(\pi x)\Big|_1^2\\ 
\Longrightarrow \quad \boxed{(u,v)_{H^1} = 0}
$$
Then, the functions $u$ and $v$ are orthogonal on $H^1(0,2)$.

In [5]:
integral, err = quadgk(x -> up(x)*vp(x), 0, 2, rtol=1e-8)

(-4.105099355129983e-15, 3.7515060282571523e-19)

### Distance Between $u$ and $v$

Let's calculate the distance between $u$ and $v$ in both spaces $L^2(0,2)$ and $H^1(0,2)$.

In [6]:
plot_umv2  = scatter(;x=x, y=broadcast(abs, u.(x) - v.(x)).^2, mode="lines+markers", fill="tozeroy", name="|u(x)-v(x)|^2")
plot_umv  = scatter(;x=x, y=u.(x) - v.(x), mode="lines+markers", name="u(x)-v(x)")

plot_upmvp2 = scatter(;x=x, y=broadcast(abs, up.(x) - vp.(x)).^2, mode="lines+markers", fill="tozeroy", name="|u'(x) -v'(x)|^2")
plot_upmvp  = scatter(;x=x, y= up.(x) - vp.(x), mode="lines+markers", name="u'(x)-v'(x)")

p3 = plot([plot_umv2, plot_umv], Layout(title="Distance Between Functions",  xaxis_title="x"))
p4 = plot([plot_upmvp2, plot_upmvp], Layout(title="Distance Between Derivative Functions",  xaxis_title="x"))

p = [p3 p4]
#relayout!(p, height=300, width=700, title_text="Functions", legend_title_text="Legend")
#p

data: [
  "scatter with fields fill, mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields fill, mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis"
]

layout: "layout with fields annotations, margin, xaxis1, xaxis2, yaxis1, and yaxis2"

#### -  $L^2(0,2)$ Space

$$ 
||u-v||_{L^2}^2 =  \int_0^2 |u-v|^2 dx =  \int_0^2 (u-v)^2 dx\\
||u-v||_{L^2}^2 =  \int_0^1 (x - \sin(\pi x))^2 dx + \int_1^2 (2-x-  \sin(\pi x))^2 dx  \\
||u-v||_{L^2}^2 =  \int_0^1 \left(x^2 - 2x\sin(\pi x) + \sin^2(\pi x)\right) dx + \int_1^2 \left(4 + x^2 + \sin^2(\pi x) -4x -4 \sin(\pi x) + 2x\sin(\pi x) \right) dx  $$

Using integral tables

$$
||u-v||_{L^2}^2 =   \left(\frac{1}{3}x^3 - \frac{2}{\pi^2}\sin(\pi x) + \frac{2}{\pi} x\cos(\pi x) + \frac{x}{2}  - \frac{1}{4\pi} \sin(2\pi x)\right)\Big|_0^1 + \left(4x + \frac{1}{3}x^3 + \frac{x}{2}  - \frac{1}{4\pi} \sin(2\pi x)  -2x^2 + \frac{4}{\pi} \cos(\pi x) + \frac{2}{\pi^2}\sin(\pi x) - \frac{2}{\pi} x\cos(\pi x)  \right)\Big|_1^2  
$$

$$
||u-v||_{L^2}^2 =   \left(\frac{1^3}{3} + \frac{2}{\pi} \cos(\pi) + \frac{1}{2} \right)+ \left(4(2) + \frac{2^3}{3} + \frac{2}{2} -(2)2^2 +\frac{4}{\pi} \cos(2\pi)  - \frac{4}{\pi} \cos(2\pi)  \right) -  \left(4 + \frac{1^3}{3} + \frac{1}{2}  -(2)1^2 +\frac{4}{\pi} \cos(\pi ) - \frac{2}{\pi} \cos(\pi)  \right) \\
||u-v||_{L^2}^2 =   \left(\frac{1}{3} - \frac{2}{\pi}  + \frac{1}{2} \right) + \left(8 + \frac{2^3}{3} + 1 - 8 \right) -  \left(4 + \frac{1}{3} + \frac{1}{2} -2 - \frac{4}{\pi} + \frac{2}{\pi}  \right) \\
||u-v||_{L^2}^2 =  \left(\frac{5}{6} - \frac{2}{\pi} \right) + \left( \frac{5}{6} - \frac{2}{\pi}\right) = \frac{5}{3} \qquad 
\Longrightarrow \qquad \boxed{||u-v||_{L^2}^2 =  1.67}
$$

In [7]:
integral1, err = quadgk(x -> broadcast(abs, u(x)-v(x))^2, 0, 2, rtol=1e-8)

(1.6666666666666665, 8.316655142337481e-9)

#### - $H^1(0,2)$ Space

$$ 
||u-v||_{H^1}^2 =  ||u-v||_{L^2}^2 + ||(u-v)'||_{L^2}^2 =  \int_0^2 |u-v|^2 dx  + \int_0^2 |(u-v)'|^2 dx 
$$

The first integral is already done, the second is

$$
||(u-v)'||_{L^2}^2 = \int_0^2 |(u-v)'|^2 dx = \int_0^1 (1 - \pi\cos(\pi x))^2 dx + \int_1^2 (-1 -\pi \cos(\pi x))^2 dx \\
\int_0^2 |(u-v)'|^2 dx = \int_0^1 \left(1 - 2\pi\cos(\pi x) + \pi^2\cos^2(\pi x)\right) dx + \int_1^2 \left(1 +2\pi \cos(\pi x)+ \pi^2 \cos^2(\pi x)\right) dx \\
||(u-v)'||_{L^2}^2 = \int_0^2 |(u-v)'|^2 dx =  \left(x - 2\sin(\pi x) + \frac{\pi^2}{2}x + \frac{\pi}{4}\sin(2\pi x) \right)\Big|_0^1  + \left(x +2\sin(\pi x)+  \frac{\pi^2}{2}x + \frac{\pi}{4}\sin(2\pi x)\right)\Big|_1^2 \\
||(u-v)'||_{L^2}^2 = \int_0^2 |(u-v)'|^2 dx =  \left(1 + \frac{\pi^2}{2} \right)  + \left(1 + \frac{\pi^2}{2}\right) =  \pi^2 +2 \qquad \Longrightarrow \qquad \boxed{||(u-v)'||_{L^2}^2  = 11.8696}
$$

In [8]:
integral2, err = quadgk(x -> broadcast(abs, up(x)-vp(x))^2, 0, 2, rtol=1e-8)

(11.869604401089356, 8.208209756332963e-8)

Finally, the distance between $u$ and $v$ in $H^1(0,2)$ is 
$$
||u-v||_{H^1}^2 =  \frac{5}{3}+ \pi^2 +2  = \frac{11}{3} + \pi^2 \qquad \Longrightarrow \qquad \boxed{||u-v||_{H^1} = 3.6792}
$$

In [9]:
umvH = sqrt(integral1 + integral2)

3.6791671704009348

## 3. T
Using the functions $u$ and $u-v$ from the previous item, verify the Cauchy-Schwarz inequality for $L^2(0,2)$ and $H^1(0,2)$.

In [10]:
plot_umumv2    = scatter(;x=x, y=broadcast(abs, u.(x).*(u.(x) - v.(x))).^2, mode="lines+markers", fill="tozeroy", name="|u(u-v)|^2")
plot_upmupmvp2 = scatter(;x=x, y=broadcast(abs, up.(x).*(up.(x) - vp.(x))).^2, mode="lines+markers", fill="tozeroy", name="|u'(u'-v')|^2")

p5 = plot([plot_u, plot_umv, plot_umumv2], Layout(title="Functions",  xaxis_title="x"))
p6 = plot([plot_up, plot_upmvp, plot_upmupmvp2], Layout(title="Derivative Functions",  xaxis_title="x"))

p = [p5 p6]
#relayout!(p, height=300, width=700, title_text="Functions", legend_title_text="Legend")
#p

data: [
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields fill, mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields mode, name, type, x, xaxis, y, and yaxis",
  "scatter with fields fill, mode, name, type, x, xaxis, y, and yaxis"
]

layout: "layout with fields annotations, margin, xaxis1, xaxis2, yaxis1, and yaxis2"

### Solution

Now, let's verify the Cauchy-Schwarz inequality in the space $L^2(0,2)$ for the previous functions, $u$ and $u-v$ 

$$
||(u, u-v)||_{L^2} \leq ||u||_{L^2} \cdot |u-v||_{L^2}  
$$

Notice that only the last term is calculated. First let's calculate the norm of the inner product between $u$ and $u-v$

$$
||(u, u-v)||_{L^2}^2 = \int_0^2 |u(u-v)|^2 dx = \int_0^1 (x(x-\sin(\pi x)))^2 dx + \int_1^2 ((2-x)(2-x-\sin(\pi x)))^2 dx \\
||(u, u-v)||_{L^2}^2 = \int_0^1 \left(x^4+x^2\sin^2(\pi x) - 2x^3\sin(\pi x)\right) dx + \int_1^2 \left((2-x)^4+(2-x)^2 \sin^2(\pi x)-2(2-x)^3\sin(\pi x) \right) dx \\
||(u, u-v)||_{L^2}^2 = \int_0^1 x^4 dx + \frac{1}{2}\int_0^1 x^2 \left(1-\cos(2\pi x) \right) dx -2 \int_0^1 x^3\sin(\pi x)dx + \int_1^2 (2-x)^4 dx  + \frac{1}{2}\int_1^2 (2-x)^2 \left(1-\cos(2\pi x) \right) dx -2 \int_1^2 (2-x)^3\sin(\pi x) dx \\
||(u, u-v)||_{L^2}^2 = \int_0^1 x^4 dx + \frac{1}{2}\int_0^1 x^2 dx - \frac{1}{2}\int_0^1 x^2 \cos(2\pi x) dx -2 \int_0^1 x^3\sin(\pi x)dx + \int_1^2 (2-x)^4 dx  + \frac{1}{2}\int_1^2 (2-x)^2  dx - \frac{1}{2}\int_1^2 (2-x)^2 \cos(2\pi x)  dx -2 \int_1^2 (2-x)^3\sin(\pi x) dx \\
||(u, u-v)||_{L^2}^2 = \frac{1}{5} + \frac{1}{6} - \frac{1}{2} \frac{1}{2\pi^2} - \frac{2}{\pi} + \frac{12}{\pi^3} + \frac{1}{5} + \frac{1}{6} - \frac{1}{2} \frac{1}{2\pi^2} - \frac{12}{\pi^3} + \frac{2}{\pi} = \frac{2}{5} + \frac{1}{3} - \frac{1}{2\pi^2} = \frac{11}{15}  - \frac{1}{2\pi^2} \qquad \Longrightarrow \qquad \boxed{||(u, u-v)||_{L^2}^2 = 0.683}
$$

Now, let's calculate the norm of $u$

$$
||u||_{L^2}^2 = \int_0^2 |u|^2 dx =  \int_0^1 x^2 dx  + \int_1^2 (2-x)^2 dx = \frac{1}{3}x^3\Big|_0^1 - \frac{1}{3} (2-x)^3\Big|_1^2 = \frac{2}{3} 
$$


To check the Cauchy-Schwarz inequality let's use it to the power of two 

$$
||(u, u-v)||_{L^2}^2 \leq ||u||_{L^2}^2 \cdot ||u-v||_{L^2}^2  \qquad \Longleftrightarrow \qquad  \boxed{0.683 \leq \frac{2}{3}(1.67) = 1.1111}
$$

In [11]:
integraluumv, err = quadgk(x -> broadcast(abs, u.(x).*(u.(x) - v.(x))).^2, 0, 2, rtol=1e-8)

(0.6826727415121643, 1.262843996041596e-12)

In [12]:
integralu2, err = quadgk(x -> broadcast(abs, u.(x)).^2, 0, 2, rtol=1e-8)

(0.6666666666666667, 5.551115123125783e-17)

In [13]:
sqrt(integraluumv) <= sqrt(integralu2) *  sqrt(umvH)

true

Now, let's verify the Cauchy-Schwarz inequality in the space $H^1(0,2)$ for the previous functions, $u$ and $u-v$ 

$$
||(u, u-v)||_{H^1} \leq ||u||_{H^1} \cdot ||u-v||_{H^1} = \left( ||u||_{L^2}^2  + ||u'||_{L^2}^2 \right)^{1/2} \left( ||u-v||_{L^2}^2  + ||(u-v)'||_{L^2}^2 \right)^{1/2}
$$

Notice that only the last term is calculated. Let's start calculating the norm of the inner product between $u$ and $u-v$

$$
||(u, u-v)||_{H^1}^2 = ||(u, u-v)||_{L^2}^2 +||(u', u'-v')||_{L^2}^2 = \int_0^2 |u (u-v)|^2 dx + \int_0^2 |u' (u'-v')|^2 dx \\
$$

Let's start calculating the inner product between the derivatives of $u$ and $u-v$
$$
\int_0^2 |u' (u'-v')|^2 dx = \int_0^1(1- \pi\cos(\pi x))^2 dx + \int_1^2 (1+\pi\cos(\pi x)) ^2dx =  \int_0^1(1- 2\pi\cos(\pi x) + \pi^2\cos^2(\pi x)) dx + \int_1^2 (1+2\pi\cos(\pi x)+\pi^2\cos^2(\pi x))dx \\
\int_0^2 |u' (u'-v')|^2 dx = \int_0^1 dx -2\pi \int_0^1 \cos(\pi x) dx + \frac{\pi^2}{2}\int_0^1(1+\cos(2\pi x)) dx + \int_1^2 dx + 2\pi\int_1^2 \cos(\pi x) dx + \frac{\pi^2}{2}\int_1^2 (1+\cos(2\pi x)) dx \\
\int_0^2 |u' (u'-v')|^2 dx = \int_0^1 dx -2\pi \int_0^1 \cos(\pi x) dx + \frac{\pi^2}{2}\int_0^1 dx + \frac{\pi^2}{2}\int_0^1\cos(2\pi x) dx + \int_1^2 dx + 2\pi\int_1^2 \cos(\pi x) dx + \frac{\pi^2}{2}\int_1^2 dx + \frac{\pi^2}{2}\int_1^2 \cos(2\pi x) dx \\
\int_0^2 |u' (u'-v')|^2 dx = 1 + \frac{\pi^2}{2} + 1 + \frac{\pi^2}{2} = 2+\pi^2 \qquad \Longrightarrow \qquad \boxed{\int_0^2 |u' (u'-v')|^2 dx = 11.87}
$$
Then,

$$
||(u, u-v)||_{H^1}^2 =  \frac{11}{15}  - \frac{1}{2\pi^2} + 2 + \pi^2 = \frac{41}{15} - \frac{1}{2\pi^2} + \pi^2 \qquad \Longrightarrow \qquad \boxed{||(u, u-v)||_{H^1}^2 = 12.552 }
$$
Now let's calculate the $u$ norm in $H^1$ 
$$
||u||_{H^|}^2 =  ||u||_{L^2}^2  + ||u'||_{L^2}^2 = \frac{2}{3} + \int_0^2 u'^2 dx = \frac{2}{3} + \int_0^1 dx + \int_1^2 dx =  \frac{2}{3} +2 = \frac{8}{3}
$$
Finally, checking the Cauchy-Schwarz inequality to the power of two 
$$
||(u, u-v)||_{H^1}^2 \leq ||u||_{H^1}^2 \cdot ||u-v||_{H^1}^2  \qquad \Longleftrightarrow \qquad \boxed{12.552 \leq \frac{8}{3}\cdot (13.536) = 36.0.97}
$$


In [14]:
integralupupmp, err = quadgk(x -> broadcast(abs, up.(x).*(up.(x) - vp.(x))).^2, 0, 2, rtol=1e-8)

(11.869604401089356, 8.208209756332963e-8)

In [15]:
sqrt(integraluumv + integralupupmp) <= sqrt(integralu2 + 2) * umvH

true

In [16]:
plot_u2  = scatter(;x=x, y=broadcast(abs, u.(x).*u.(x)), mode="lines+markers", fill="tozeroy", name="|u|^2")
plot_up2 = scatter(;x=x, y=broadcast(abs, up.(x).*up.(x)), mode="lines+markers", fill="tozeroy", name="|u'|^2")

plot([plot_u2, plot_up2], Layout(title="Functions",  xaxis_title="x"))

data: [
  "scatter with fields fill, mode, name, type, x, and y",
  "scatter with fields fill, mode, name, type, x, and y"
]

layout: "layout with fields margin, title, and xaxis"

## 4. T

Show that, if $f(x,y)$ is a continous function defined in $\mathbb{R}^2$ such that

$$ \iint_\Omega f(x,y) \; dx dy = 0 $$

for all rectangle $\Omega \in \mathbb{R}^2$, the $f(x,y)=0$ for all $(x,y)$.

### Solution

Let's suppose a continous function $f$ such that $f(x_0,y_0)=L>0$ for some $(x_0,y_0) \in\Omega$, and $\iint_{\Omega} f(x,y) dx dy = 0$. 

Since $f$ is continuous, for all $\epsilon > 0$, there exist $\delta$ such that if $|(x,y)-(x_0,y_0)|<\delta$ then $|f(x,y)-L|<\epsilon$. It is valid for any $\epsilon >0$. In particular, taking $\epsilon=L/2$ the $f(x,y)$ inequality takes the following form

$$-\epsilon<f(x,y)-L<\epsilon \\
-\frac{L}{2} < f(x,y)-L< \frac{L}{2}\\
-\frac{L}{2}+L < f(x,y) < \frac{L}{2}+L\\
\frac{L}{2} < f(x,y) < \frac{3L}{2}$$

integrating the last inequality over $\Omega$

$$\iint_{\Omega} \frac{L}{2} \;dxdy< \iint_{\Omega}  f(x,y) \;dxdy< \iint_{\Omega} \frac{3L}{2} \; dxdy$$

then

$$ \iint_{\Omega}  f(x,y) \;dxdy > \iint_{\Omega} \frac{L}{2} \;dxdy \geq \int_{y_0-\delta}^{y_0+\delta} \int_{x_0-\delta}^{x_0+\delta} \frac{L}{2} \; dx dy = \frac{L}{2} (2\delta) (2\delta) = 2 L \; \cdot \; \delta^2 >0$$ 

which is a contradiction. Similarly, the case when $f(x_0,y_0)<0$ is can discussed. 

Hence, $f(x,y)$ must be $0$ $\left(f(x,y)=0\right)$ for all $(x,y)\in\Omega$.

## 5. T
Let $X$ and $Y$ Banach spaces and $A:\;X\;\to\;Y$ a linear operator. Prove that the following  statement are equivalent

- $(i)$ $A$ is enclosed, i.e., exits a constat $C>0$ such that
$$ ||Au||_Y \leq C||u||_X, \qquad \forall u\in X$$

- $(ii)$ $A$ is continous, i.e.
$$\text{Si } u_n\to u \; \text{cuando } n\to \infty, \; \text{entonces }Au_n \to Au \text{cuando } n\to\infty$$

### Solution 
#### $(i)\rightarrow(ii)$

Let $(x_n)\subseteq X$ be a convergent sequence with limit $x \in X$, then $x_n\rightarrow x$ when $n\rightarrow\infty$

$$||Ax_n-Ax||_Y=||A(x_n-x)||_Y=||A||||x_n-x||_X\rightarrow 0, \; \text{because } x_n\rightarrow x \quad \Rightarrow \quad Ax_n\rightarrow Ax, \; \text{when }  n\rightarrow\infty$$

#### $(ii)\rightarrow(i)$

$A$ is continuous in all point, it implies $A$ is continous in $0$  .

$(*) :$ For all sequences $(x_n)\subseteq X$ with $x_n\rightarrow 0$ when $n\rightarrow\infty$, we have $Ax_n\rightarrow 0$ when $n\rightarrow\infty$ 

$(**) :$ Then, there is some $\delta > 0$ such that $||Ax||_Y<1$ for all $x \in X$ with $||x||_X<\delta$

Claim: $(*) \Rightarrow (**)$

#### Proof of the claim:

$\neg(**)\Rightarrow$ for all $n\in N$ we find $x_n \in X$ with norm $||x_n||_X<\frac{1}{n}$ and $||Ax_n||_Y\geq 1$ $\Rightarrow \neg(*) $

$$\frac{||Ax||_Y}{||x||_X} = \frac{||Ax||_Y\frac{\delta}{2||x||_X}}{||x||_X \frac{\delta}{2||x||_X}} = \frac{\left|\left|A\left(\frac{\delta x}{2||x||_X}\right)\right|\right|_Y}{\left|\left|\frac{\delta x}{2||x||_X}\right|\right|_X}\leqslant\frac{2}{\delta} \quad \Rightarrow\quad ||A||=sup\left\{\frac{||Ax||_Y}{||x||_X} : x\in X,  x \neq 0 \right\}\leqslant\frac{2}{\delta}<\infty$$ 

Hence, $A$ is bounded.

You can find this homework in the github repository [saguileran/NAII](https://github.com/saguileran/NAII).